config/config.go

```diff
@@ -694,7 +694,7 @@ func getStakingSigner(v *viper.Viper) (*bls.SecretKey, error) {
 		return nil, fmt.Errorf("couldn't create path for signing key at %s: %w", signingKeyPath, err)
 	}
 
-	keyBytes := bls.SecretKeyToBytes(key)
+	keyBytes := key.ToBytes()
```

Comment:
I'll have to change this eventually to fully abstract the signer. I'm open to ideas on how best to do this

Reply:
Yeah, looking at the code again, signer creation would be specific. A `LocalSigner` should actually write the key to disk before returning from a constructor. We could have an `EphemeralSigner` that's similar but doesn't save the key and is used for testing. 

With that said, I would probably create a 
```go
func initializeSigner(signerConfig) (Signer) { /* ... */ }
```
and deal with saving or not saving there. It should not be dealt with here IMO. 

This is all for a future PR though